src/main/java/com/treasure/hunt/game/Turn.java

```diff
@@ -28,29 +29,72 @@
      */
     private Hint hint;
     /**
-     * The {@link SearchPath} the {@link com.treasure.hunt.strategy.searcher.Searcher} did.
+     * The {@link SearchPathPrototype} the {@link com.treasure.hunt.strategy.searcher.Searcher} did.
      */
     private SearchPath searchPath;
     /**
      * The current location of the treasure.
      */
-    private Point treasureLocation;
+    private Treasure treasure;
+
+    public Turn(Hint hint, SearchPath searchPath, Coordinate treasure) {
+        this.hint = hint;
+        this.searchPath = searchPath;
+        this.treasure = new Treasure(treasure);
+    }
 
     /**
      * @return a list of all geometryItems of this.
      */
-    public List<GeometryItem<?>> getGeometryItems(Point movementStart) {
+    public List<GeometryItem<?>> getGeometryItems() {
         List<GeometryItem<?>> output = new ArrayList<>();
 
         if (hint != null) {
             output.addAll(getHintGeometries());
         }
 
         if (searchPath != null) {
-            output.addAll(getSearchPathGeometries(movementStart));
+            output.addAll(getSearchPathGeometries());
+        }
+
+        if (treasure != null) {
+            output.addAll(getTreasureGeometries());
+        }
+
+        output.sort(Comparator.comparingInt(a -> a.getGeometryStyle().getZIndex()));
+
+        return output;
+    }
+
+    public void unselect() {
+        if (hint != null) {
+            hint.setSelected(false);
+        }
+
+        if (searchPath != null) {
+            searchPath.setSelected(false);
+        }
+
+        if (treasure != null) {
+            treasure.setSelected(false);
         }
+    }
```

Comment:
What has this to do here

Reply:
outdated, since selected field will be removed